### PR TITLE
[release-1.26] test: upgrade golangci/golangci-lint-action to v1.54

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

This is a cherry-pick of #2073 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .github/workflows/static.yaml [hunk 1]
```diff
@@ -15,5 +15,5 @@ jobs:
             - name: Run linter
               uses: golangci/golangci-lint-action@v3
               with:
-                  version: v1.51
-                  args: -E=gofmt,deadcode,unused,varcheck,ineffassign,revive,misspell,exportloopref,asciicheck,bodyclose,depguard,dogsled,durationcheck,errname,forbidigo -D=staticcheck --timeout=30m0s
+                  version: v1.54
+                  args: -E=gofmt,unused,ineffassign,revive,misspell,exportloopref,asciicheck,bodyclose,depguard,dogsled,durationcheck,errname,forbidigo -D=staticcheck --timeout=30m0s
```
Comment (MartinForReal) [inline .github/workflows/static.yaml:19]: I'm wondering if we need to put all of these linters to config file.

**Review by andyzhangx — APPROVED** [review]
/ok-to-test

File: .github/workflows/static.yaml (hunk 1 above)
Comment (umagnus) [reply .github/workflows/static.yaml:19]: I think these checks will not affect normal code. In special cases, we can also ignore the corresponding linters.

**Pull request merged**